blaze-server/src/main/scala/org/http4s/blaze/server/Http1ServerStage.scala

```diff
@@ -229,15 +229,13 @@ private[blaze] class Http1ServerStage[F[_]](
               F.runCancelable(action) {
                 case Right(()) => IO.unit
                 case Left(t) =>
-                  IO(logger.error(t)(s"Error running request: $req")).attempt *> IO(
-                    closeConnection()
-                  )
+                  IO(logger.error(t)(s"Error running request: $req")).attempt *>
+                    IO.delay(cancelToken.set(None)) *>
```

Comment:
The `closeConnection()` is cancel previous request. Probably after successful request processing, nothing is happening in the case of that canceling. But I think it's better to avoid that canceling at all.